contrib/corrosion-cmake/CMakeLists.txt

```diff
@@ -116,3 +120,30 @@ set(RUST_VENDOR_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../rust_vendor")
 # This needs to be done for each project using set_property(INTERFACE_CORROSION_CARGO_FLAGS)
 configure_file("config.toml.in" "config.toml" @ONLY)
 set (RUST_CONFIG_FILE "${CMAKE_CURRENT_BINARY_DIR}/config.toml" CACHE INTERNAL "Rust config")
+
+function(clickhouse_import_crate)
+    # This is a workaround for Corrosion case sensitive build type matching in
+    # _generator_add_cargo_targets(), that leads to different paths in
+    # IMPORTED_LOCATION and real path of the library.
+    #
+    # It uses CMAKE_CONFIGURATION_TYPES and $<CONFIG>, so here we preserve the
+    # case of ${CMAKE_BUILD_TYPE} in ${CMAKE_CONFIGURATION_TYPES}.
+    if ("${CMAKE_BUILD_TYPE_UC}" STREQUAL "DEBUG")
+        set(CMAKE_CONFIGURATION_TYPES "${CMAKE_BUILD_TYPE};release")
+    else()
+        set(CMAKE_CONFIGURATION_TYPES "${CMAKE_BUILD_TYPE};debug")
+    endif()
+
+    if (CMAKE_BUILD_TYPE_UC STREQUAL "DEBUG")
+        set(profile "")
+    else()
+        if (ENABLE_THINLTO)
+            set(profile "release-thinlto")
+        else()
+            set(profile "release")
+        endif()
+    endif()
+
+    corrosion_import_crate(${ARGN} NO_STD PROFILE ${profile} CRATE_TYPES staticlib LOCKED FLAGS --offline NO_DEFAULT_FEATURES)
```

Comment:
I guess we can clean up some crates from `rust-vendor` after `NO_DEFAULT_FEATURES`?

Reply:
Probably, but I don't know how to generate the lock files without default features (or just the one enabled for delta-kernel).